pkg/tsdb/elasticsearch/response_parser.go

```diff
@@ -101,10 +110,11 @@ func parseResponse(ctx context.Context, responses []*es.SearchResponse, targets
 			trimDatapoints(queryRes, target)
 
 			result.Responses[target.RefID] = queryRes
+			span.End()
```

Comment:
I think this should be one line lower to be outside the `if` `else` case, so the span will be ended every time.

Reply:
Yes, good catch! Fixed in https://github.com/grafana/grafana/pull/74750/commits/1509a01bb63bc85fccb069bdb2350ed20475f40a